src/main/java/at/hannibal2/skyhanni/features/inventory/chocolatefactory/ChocolateFactoryTooltipStray.kt

```diff
@@ -16,10 +16,11 @@ object ChocolateFactoryTooltipStray {
      * REGEX-TEST: §7You gained §6+2,465,018 Chocolate§7!
      * REGEX-TEST: §7gained §6+30,292 Chocolate§7!
      * REGEX-TEST: §7§6+36,330 Chocolate§7!
+     * REGEX-TEST: §9Rabbit§7, so you received §655,935,257
      */
     private val chocolateGainedPattern by ChocolateFactoryAPI.patternGroup.pattern(
         "rabbit.stray",
-        "(?:§.)+(?:You )?(?:gained )?§6\\+(?<amount>[\\d,]+) Chocolate§7!"
+        "(?:§.)?+(?:Rabbit§7, so )?(?:[Yy]ou )?(?:gained |received )?§6\\+?(?<amount>[\\d,]+)(?:\$| Chocolate§7!)"
```

Comment:
pretty sure you shouldnt be escapign the $ here

Reply:
It's being string-escaped, not truly escaped in the pattern itself. It getting prepended at all was a result of pasting the regex in from r101, if I had to guess.

    private val p1 by patternGroup.pattern(
        "testpatterngroup.pattern",
        "(?:§.)?+(?:Rabbit§7, so )?(?:[Yy]ou )?(?:gained |received )?§6\\+?(?<amount>[\\d,]+)(?:\$| Chocolate§7!)"
    )
    private val p2 by patternGroup.pattern(
        "testpatterngroup.pattern2",
        "(?:§.)?+(?:Rabbit§7, so )?(?:[Yy]ou )?(?:gained |received )?§6\\+?(?<amount>[\\d,]+)(?:$| Chocolate§7!)"
    )
    
        val testString = "§9Rabbit§7, so you received §655,935,257";
        ChatUtils.chat("P1 match? " + (p1.matches(testString).toString()) + "P2 match? " + (p2.matches(testString).toString()))

![image](https://github.com/user-attachments/assets/05b9aa43-b555-4a39-b596-8bc2cac8b0ed)

Reply:
id still prefer if we use the pattern that does match in regex 101 as that is the kinda the point of the regex tests